service/view_building_coordinator.cc

```diff
@@ -432,6 +435,51 @@ future<canonical_mutation> view_building_coordinator::remove_built_view(const gr
     co_return canonical_mutation(std::move(mut));
 }
 
+static bool contains_range(const dht::token_range_vector& ranges, const dht::token_range& range) {
+    return std::find(ranges.cbegin(), ranges.cend(), range) != ranges.cend();
+}
+
+future<std::vector<mutation>> view_building_coordinator::get_migrate_tasks_mutations(const group0_guard& guard, table_id table_id, std::optional<locator::tablet_replica> abandoning_replica, std::optional<locator::tablet_replica> pending_replica, dht::token_range range) {
+    auto state = co_await load_coordinator_state();
+    if (!state.tasks.contains(table_id)) {
+        co_return std::vector<mutation>();
+    }
+
+    std::vector<mutation> updates;
+    for (auto& [view, tasks]: state.tasks[table_id]) {
+        if (abandoning_replica && pending_replica) {
+            view_building_target target{abandoning_replica->host, abandoning_replica->shard};
+            if (tasks.contains(target) && contains_range(tasks[target], range)) {
+                if (_per_host_processing_range.contains(target) && _per_host_processing_range[target] == range) {
+                    co_await abort_work(target.host, target.shard);
```

Comment:
It's a bit surprising that a function whose name starts with `get_` performs actions like aborting work on remote shard...